optaplanner-wb-screens/optaplanner-wb-solver-editor/optaplanner-wb-solver-editor-backend/src/main/java/org/optaplanner/workbench/screens/solver/backend/server/PathUtil.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.optaplanner.workbench.screens.solver.backend.server;
+
+import org.uberfire.backend.vfs.Path;
+
+public class PathUtil {
+
+    public static int getUriLength(Path path) {
+        String s = path.toURI();
+        if ("/".equals(path.getFileName())) {
```

Comment:
Looking at the tests, why not just check if the last character of URI String is "/" ?

Reply:
Only reason is that I'm not 100% sure that works. Probably does. Maybe it is even safer, since it feels weird the file name is "/" for folders, I think they should have the folder name in them.

Reply:
Maybe it is safer to use the last character of uri. I'll change it.